src/metrics.rs

```diff
@@ -102,13 +102,24 @@ impl<'a> Serialize for FuncSpace<'a> {
 
 impl<'a> FuncSpace<'a> {
     fn new<T: Getter>(node: &Node<'a>, code: &'a [u8], kind: NodeKind) -> Self {
+        let mut start_position = node.start_position().row;
```

Comment:
I'm not happy this change because start_line & end_line have not their expected values or maybe I missed something.
Suppose you have a function like `fn foo() -> u32 { 1 }`, with your change it's location (i.e. start/end_line) will be 0, 0.

Reply:
Oops, you're completely right, it should be fixed now, thanks a lot!

Reply:
You could probably just check if the unit is empty (i.e. no children) and then set position to 0 in this only case, I suppose in this case it's acceptable.

Reply:
> You could probably just check if the unit is empty (i.e. no children) and then set position to 0 in this only case, I suppose in this case it's acceptable.

Okay, I'm going to change my implementation, thanks!

> FYI, I always inject an EOL as last char: https://github.com/mozilla/rust-code-analysis/blob/master/src/tools.rs#L39 because some parsers expect to have one.

Now I know because I got (1,1) when I tried with an empty file. But is it always right to add an `EOL` at the end of a file? Because a user, not knowing of this feature, could be confused imho.

Reply:
The EOL injection is just a workaround (only when no last eol).
In all firefox source code, I found only one cpp file with this final eol and in this case the parsing was kinda broken.

Reply:
I see, thanks, so I will implement what you said above without changing anything else

Reply:
Done